text/0000-process-type-arguments.md

```diff
@@ -0,0 +1,221 @@
+
+Instead, at execution time, the lifecycle should allow a user to specify the arguments to append to an existing process type.
+
+Feedback from users in buildpacks slack and in issues, indicates that should in fact be the default behavior:
+- https://github.com/buildpacks/lifecycle/issues/280
+- https://github.com/buildpacks/pack/issues/468#issuecomment-593139689
+
+## Problem 2: Arguments for Shell Processes
+Currently, arguments defined in a shell process are treated as arguments to bash, rather than as arguments to the command.
+Given a process:
+```
+[[process]]
+command = "<CMD>"
+args = ["ARG1", "ARG2"]
+```
+the `launcher` currently executes `bash -c "<CMD>" "ARG1" "ARG2"`. This makes it hard to use arguments intuitively in a shell process.
+For example, the following does not print `hello world`.
+```
+[[process]]
+command = "echo"
+args = ["hello", "world"]
+```
+Instead, users must construct a string command that includes arguments
+```
+[[process]]
+command = "echo hello world"
+args = []
+```
+This already causes confusion for users (https://github.com/buildpacks/lifecycle/issues/281). The problem is compounded if we allow users to supply additional arguments. We cannot reliably append arguments to the `command` string without introducing tokenization or shell parsing errors. Constructing a `command` that accepts args in the way they are currently provided to `bash` is confusing and difficult (buildpack authors likely will not do this correctly).
+
+This RFC proposes changes to the `launcher`'s handling of args to make the execution of the following process
+```
+[[process]]
+command = "<CMD>"
+args = ["ARG1", "ARG2"]
+```
+produce results equivalent to the execution of `"<CMD>" "<ARG1>" "<ARG2>"` in a shell
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Terminology
+* shell process - a process with `direct=false`
+
+## Philosophy
+The goal is to make the runtime usage of a CNB container as similar as possible to the runtime usage on a non-CNB container, without removing functionality or violating assumptions that have been firmly established in the buildpack community (e.g. defaulting to `web`).
+
+Because the `launcher` must occupy the container `entrypoint`, we will make the `launcher` a multicall binary that derives the intended process-type, or lack there of, from `argv0`. This allows for intuitive use of both `entrypoint` and `cmd`: `entrypoint` will be used to select a process and `cmd` will be used to supply additional argument to that process.
+
+## Behavior
+
+### Exporter
+When `exporter` creates an app image it will create one symlink per buildpack-contributed process type. The symlink files will be named `/cnb/process/<process-type>` and will point at the launcher.
```

Comment:
I couldn't find anything in the spec that says `[[process]] type = <process-type>` has to be a certain format. Is that an implementation detail somewhere perhaps?